dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CheckUrlInterface.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import java.util.Set;
+
+/**
+ * <p>Determines the openness of a set of urls. An open URL:</p>
+ *
+ * <ul>
+ *     <li>Has a protocol recognized out of the box by Java. Note that this does NOT include S3
+ *     and GS URIs</li>
+ *     <li>Is not a local file; caveat to above, as file:// is a protocol recognized by Java</li>
+ *     <li>Is accessible without authentication</li>
+ * </ul>
+ *
+ */
+public interface CheckUrlInterface {
```

Comment:
STRONG OPINION WARNING:
So... I'm gonna lobby against appending `Interface` to our interface names.  Ideally, most code will operate on/with interface references, and IMHO, the repeated `Interface` verbiage clouds the code and provides very little value.  We know it's an interface, the IDE knows it's an interface, no need to belabor the point. :)  Similar feelings about `Impl`. An example of `Interface` and `Impl`-free naming:

```
interface Foo
abstract class BaseFoo implements Foo
class UsefulFoo extends BaseFoo
class ExtremeFoo extends BaseFoo
class CompositeFoo implements Foo
class LoggingFoo implements Foo
```
And most everywhere that uses a `Foo` is passed a `Foo` reference, rather than a reference of an implementing class type.

Reply:
I prefer to not have `Interface` myself, but if my grep is correct, we have 12 public interfaces declared in the web service, 9 of whose names end with `Interface`. I'd rather stick to the existing pattern and change it globally on team consensus rather than doing a one-off here.